pyfr/__main__.py

```diff
@@ -187,33 +199,41 @@ def process_partition(args):
             raise RuntimeError('No partitioners available')
 
     # Partition the mesh
-    mesh, rnum, part_soln_fn = part.partition(mesh)
+    mesh, rnum, part_soln_fn = part.partition(mesh, progress)
 
-    # Prepare the solutions
-    solnit = (part_soln_fn(NativeReader(s)) for s in args.solns)
+    # Write the repartitioned mesh file
+    with progress.start('Write mesh'):
+        write_pyfrms(os.path.join(args.outd, os.path.basename(args.mesh)),
+                     mesh)
+
+    # Write out the renumbering table
+    if args.rnumf:
+        with progress.start('Write renumbering table'):
+            print('etype,pold,iold,pnew,inew', file=args.rnumf)
 
-    # Output paths/files
-    paths = it.chain([args.mesh], args.solns)
-    files = it.chain([mesh], solnit)
+            for etype, emap in sorted(rnum.items()):
+                for k, v in sorted(emap.items()):
+                    print(etype, *k, *v, sep=',', file=args.rnumf)
 
-    # Iterate over the output mesh/solutions
-    for path, data in zip(paths, files):
-        # Compute the output path
-        path = os.path.join(args.outd, os.path.basename(path.rstrip('/')))
+    # Repartition any solutions
+    if args.solns:
+        with progress.start_with_bar('Repartition solutions') as pb:
+            pb.set_end(len(args.solns))
 
-        # Save to disk
-        write_pyfrms(path, data)
+            for ipath in args.solns:
+                # Compute the output path
+                opath = os.path.join(args.outd, os.path.basename(ipath))
 
-    # Write out the renumbering table
-    if args.rnumf:
-        print('etype,pold,iold,pnew,inew', file=args.rnumf)
+                # Save to disk
+                write_pyfrms(opath, part_soln_fn(NativeReader(ipath)))
 
-        for etype, emap in sorted(rnum.items()):
-            for k, v in sorted(emap.items()):
-                print(etype, *k, *v, sep=',', file=args.rnumf)
+                pb.advance_by(1)
 
 
 def process_export(args):
+    # Create a suitable progress tracker
+    progress = ProgressSequence() if args.progress else NullProgressSequence()
```

Comment:
Is it worth making `args.progress` an argument to the ProgressSequence class? It could simplify these if statements

Reply:
This should now be resolved.